### ISSUE TITLE
WD MyNet 750 WiFi fails between 10 and 20 days of uptime.

### ISSUE THREAD

**Issue by apassiou** [ericpaulbishop/gargoyle#932]
Issue goes back as far as Gargoyle version 1.10 (and possibly earlier), verified on 3 different units. Strangely issue is not present in the My Net 600 version of the routers.

Behavior is as follows. Router is up and functioning. After about a 10 days (usually closer to 17 days) devices connected to wifi of the router will not be able to route to internet. 

Things I have tried:
- Extend client timeout (in GUI) from default 12 hours (bit too short for default eh?) to 72 hours. No change.
- checked dmesg. Nothing of use there, just bootup messages.
- checked logread. Nothing there either around the timeframe of the issue occuring. 
- manually **wifi down** then **wifi up**, this does fix it. For another 2 weeks or so.

So my current workaround is to crontab an nightly script to bring down wifi, wait 10 seconds and then bring it back up. Its working. But ideally this issue is resolved. Its been around for at least 4 years. Since its not happening on N600 routers Im guessing its something tied to the chipset used in N750.  Something locks up (buffer issue?) and routing breaks.

Im not sure what other useful info I can provide as logs dont seem to mention anything about it.

For those that need workaround:

```
#!/bin/sh
wifi down
sleep 10
wifi up
```

I made my script in /tmp/wifi_workaround.sh and added it in crontab to run every night when everyone is sleeping.

**Comment by ronniehd** [comment]
Hi, do you have guest network enabled? I used to have the same issue but only with 2.4GHz network since Gargoyle version 1.9.x i think. Try disabling it and test. Is what worked for me.
I have a TP-Link archer c7 v2 that i recently upgraded to 1.13.0 with guest network enabled, 9 days 13 hours uptime so I can't tell if the same bug persist but I'll report back after 20+ days uptime.

**Comment by apassiou** [comment]
No I dont have guest network enabled.

I loaded OpenWRT latest build and have had it up:

> OpenWrt 21.02.0-rc3, r16172-2aba3e9784
>  -----------------------------------------------------
> root@OpenWrt:~# uptime
>  12:57:29 up 47 days, 23:59,  load average: 0.00, 0.00, 0.00

No issue. So I think its specific to Gargoyle.

**Comment by apassiou** [comment]
I think I may have found the issue. Will need a lot of testing (since it can take a few weeks for a failure to happen). 

I believe the issue may be related to TX power, by default if Region is set to 00-World max TX power is 20dbm. If Region is set to US then max TX is 22dbm.

I noticed that the issue on my Mynet750 set to 00-World (because I never bothered to change it), and with only 2.4Ghz powered on had no issues (as it by default maxed out at 20dbm) as you can see in my post above 47 days+ and no issue.

On another mynet 750 I have I changed the value to US as region. Which defaulted to 22dBm. WiFi failed withing 24 hours. I also have 5Ghz on on this one. So something is going on with power draw, this particular router (and perhaps WiFi chip) bugs out when it needs to draw too much power.

I am guessing that 2.4Ghz by itself (with 5.0Ghz off) should do fine with 20dBm. But I am afraid that with 5.0Ghz on 20dBm will still be too much. Thats the theory.

Right now I set both 2.4Ghz and 5Ghz to 21dBm (1 notch down from max on USA setting). Will report back results.

**Comment by lantis1008** [comment]
I'm interested to hear your findings and whether this is applicable to other devices.
I don't think we would ever impose a maximum db limit (that's up to the user), but it would certainly be an interesting piece of advice that can be included in the knowledge bank.

**Comment by apassiou** [comment]
> I'm interested to hear your findings and whether this is applicable to other devices.
> I don't think we would ever impose a maximum db limit (that's up to the user), but it would certainly be an interesting piece of advice that can be included in the knowledge bank.

Yeah Ill report back. I wasnt saying to impose upper limit, I was saying to default to something like 18, with users able to go higher if they wish.

**Comment by apassiou** [comment]
Just a small update. 7 days and so far so good. Previously it would take up to 20 days to fail so Im not celebrating yet. But ill update again in a week or two.

**Comment by ronniehd** [comment]
Hi, i think this might be the same issue im now experiencing in my archer c7 v2. I did change default 00-world to US. After 7+ days it gets unresponsive, I can't login nor ssh into it, a reboot fixes it until the next 7+ days. I was thinking in downgrading to v1.12 but ill give it one more chance by updating 1.13 firmware and changing back to 00-world.  Thanks!

**Comment by apassiou** [comment]
44 days later and it happened again. Definitely a much longer time before occurrences. So I am knocking down transmission power down to 18... will report back.

**Comment by lantis1008** [comment]
1 month ok?

**Comment by apassiou** [comment]
Yes ok so far at 18.

**Comment by lantis1008** [comment]
It may be worth trying the latest test images (1.15).
Otherwise please close this issue, thanks for your investigation
